Functions/xASL_io_SplitASL.m

```diff
@@ -184,22 +194,81 @@ function xASL_io_SplitASL(inPath, iM0, iDummy)
             warning('After removing M0 volume and Dummy scans, no even number of volumes (control-label)');
         end
 
-        %% Save ASL4D NIfTI
+        %% 6. Save ASL4D NIfTI
         xASL_io_SaveNifti(BackupName,ASLname,tIM(:,:,:,ASLindices),[],false);
-
-        %% Copy sidecars
+        
+        %% 7. Split relevant JSON parameters/arrays
+        [jsonM0, jsonASL] = xASL_io_SplitASL_SplitJSON(BackupJSONPath, iM0, ASLindices);
+        
+        %% 8. Copy sidecars
         if exist(BackupMATPath,'file') && bCreateM0
             xASL_Copy(BackupMATPath, M0MATPath);
         end
         if exist(BackupMATPath,'file') && ~strcmp(BackupMATPath, ASLMATPath)
             xASL_Copy(BackupMATPath, ASLMATPath);
         end
-        if exist(BackupJSONPath,'file') && bCreateM0
-            xASL_Copy(BackupJSONPath, M0JSONPath);
+        if exist('jsonM0','var') && bCreateM0
+            spm_jsonwrite(M0JSONPath, jsonM0);
+        end
+        if exist('jsonASL','var') && ~strcmp(BackupJSONPath, ASLJSONPath)
+            spm_jsonwrite(ASLJSONPath, jsonASL);
         end
-        if exist(BackupJSONPath,'file') && ~strcmp(BackupJSONPath, ASLJSONPath)
-            xASL_Copy(BackupJSONPath, ASLJSONPath);
+        
+    end
+
+end
+
+
+%% xASL_io_SplitASL_SplitJSON
+function [jsonM0, jsonASL] = xASL_io_SplitASL_SplitJSON(BackupJSONPath, indicesM0, indicesASL)
+
+    % Make sure that we have column arrays
+    if size(indicesM0,2)>size(indicesM0,1)
+        indicesM0 = indicesM0';
+    end
+    if size(indicesASL,2)>size(indicesASL,1)
+        indicesASL = indicesASL';
+    end
+
+    % Load backup JSON
+    if exist(BackupJSONPath,'file')
+        jsonStruct = spm_jsonread(BackupJSONPath);
+        
+        % Define M0 & ASL fallback JSONs
+        jsonM0 = jsonStruct;
+        jsonASL = jsonStruct;
+        
+        % Fields which should be split
+        splitFields = {'EchoTime', 'FlipAngle', 'RepetitionTimePreparation', 'PostLabelingDelay', 'VascularCrushingVENC', 'LabelingDuration'}';
+        
+        % Iterate over JSON fields
+        jsonFields = fieldnames(jsonStruct);
+        for iField = 1:numel(jsonFields)
+            % Check if fieldname is in splitFields list
+            if ismember(jsonFields{iField,1},splitFields)
+                % Check if field has to be split
+                if numel(jsonStruct.(jsonFields{iField,1}))>1
+                    % Compare array length
+                    if (numel(indicesM0)+numel(indicesASL))==numel(jsonStruct.(jsonFields{iField,1}))
```

Comment:
Good to have this condition, but we have to adapt it, because we have incides for ASL, M0, and Dummy - these three together give the full-set.

Reply:
Thinking about how to address this, but adding the Dummy indices (dynamics to be thrown out) to the call of this function and expanding this equation here with numel(indicesDummy) seems like the best solution.

Reply:
That's an easy fix :)

Reply:
OK, are you doing it, because I have realized that I can fix it and then quickly test it at the same time :)